ui/e2e/src/pages/DashboardPage.ts

```diff
@@ -214,35 +211,67 @@ export class DashboardPage {
     await panelEditor.isClosed();
   }
 
-  getPanel(panelName: string) {
+  /**
+   * If the passed value is already a panel, return it. Otherwise, look the panel
+   * up by name. Useful for internal helper functions that can support taking
+   * a name or a panel.
+   */
+  private getPanelByNameOrSelf(panelNameOrPanel: PanelNameOrPanel) {
+    // If the passed value is already a panel, return it. Useful for internal
+    // helper functions that want to support taking a name or a panel.
+    if (typeof panelNameOrPanel !== 'string') {
+      return panelNameOrPanel;
+    }
+
+    return this.getPanel(panelNameOrPanel);
+  }
+
+  /**
+   * Get a panel by name.
+   */
+  getPanel(panelName: string): Panel {
     const container = this.panels.filter({
       has: this.page.getByRole('heading', { name: panelName }),
     });
     return new Panel(container);
   }
 
-  async editMarkdownPanel(panelName: string, { name, groupName }: EditMarkdownPanelConfig) {
-    const panel = this.getPanel(panelName);
```

Comment:
I've been unhappy with this helper for a while, so I cleaned it up and replaced it with the more generic `editPanel` while I was touching it for the tests in this PR.